dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -1626,3 +1627,112 @@ HRESULT WINAPI CFSFolder::CallBack(IShellFolder *psf, HWND hwndOwner, IDataObjec
 
     return Shell_DefaultContextMenuCallBack(this, pdtobj);
 }
+
+static HBITMAP DoLoadPicture(LPCWSTR pszFileName)
+{
+    // create stream from file
+    HRESULT hr;
+    IStream *pStream = NULL;
+    hr = SHCreateStreamOnFileEx(pszFileName, STGM_READ, FILE_ATTRIBUTE_NORMAL,
+                                FALSE, NULL, &pStream);
+    if (FAILED(hr))
+        return NULL;
+
+    // load the picture
+    HBITMAP hbm = NULL;
+    IPicture *pPicture = NULL;
```

Comment:
CComPtr may be appropriate for pStream and pPicture.

Reply:
OK.